lib/cylc/network/server.py

```diff
@@ -0,0 +1,673 @@
+        return self.public_priv
+
+    @authorise(Priv.CONTROL)
+    @ZMQServer.expose
+    def clear_broadcast(
+            self, point_strings=None, namespaces=None, cancel_settings=None):
+        """Clear settings globally, or for listed namespaces and/or points.
+
+        Return a tuple (modified_settings, bad_options), where:
+        * modified_settings is similar to the return value of the "put" method,
+          but for removed settings.
+        * bad_options is a dict in the form:
+              {"point_strings": ["20020202", ..."], ...}
+          The dict is only populated if there are options not associated with
+          previous broadcasts. The keys can be:
+          * point_strings: a list of bad point strings.
+          * namespaces: a list of bad namespaces.
+          * cancel: a list of tuples. Each tuple contains the keys of a bad
+            setting.
+        """
+        return self.schd.task_events_mgr.broadcast_mgr.clear_broadcast(
+            point_strings, namespaces, cancel_settings)
+
+    @authorise(Priv.CONTROL)
+    @ZMQServer.expose
+    def dry_run_tasks(self, items, check_syntax=True):
+        """Prepare job file for a task.
+
+        items[0] is an identifier for matching a task proxy.
+        """
+        if not isinstance(items, list):
+            items = [items]
+        self.schd.command_queue.put(('dry_run_tasks', (items,),
+                                    {'check_syntax': check_syntax}))
+        return (True, 'Command queued')
+
+    @authorise(Priv.CONTROL)
+    @ZMQServer.expose
+    def expire_broadcast(self, cutoff=None):
+        """Clear all settings targeting cycle points earlier than cutoff."""
+        return self.schd.task_events_mgr.broadcast_mgr.expire_broadcast(cutoff)
+
+    @authorise(Priv.READ)
+    @ZMQServer.expose
+    def get_broadcast(self, task_id=None):
+        """Retrieve all broadcast variables that target a given task ID."""
+        return self.schd.task_events_mgr.broadcast_mgr.get_broadcast(task_id)
+
+    @authorise(Priv.IDENTITY)
+    @ZMQServer.expose
+    def get_cylc_version(self):
+        """Return the cylc version running this suite."""
+        return CYLC_VERSION
+
+    @authorise(Priv.READ)
+    @ZMQServer.expose
+    def get_graph_raw(self, start_point_string, stop_point_string,
+                      group_nodes=None, ungroup_nodes=None,
+                      ungroup_recursive=False, group_all=False,
+                      ungroup_all=False):
+        """Return raw suite graph."""
+        # Ensure that a "None" str is converted to the None value.
+        if stop_point_string is not None:
+            stop_point_string = str(stop_point_string)
+        return self.schd.info_get_graph_raw(
+            start_point_string, stop_point_string,
+            group_nodes=group_nodes,
+            ungroup_nodes=ungroup_nodes,
+            ungroup_recursive=ungroup_recursive,
+            group_all=group_all,
+            ungroup_all=ungroup_all)
```

Comment:
We really need to do something about this. Not part of this PR though!

Reply:
Yep it's pretty horrible.